source/material_model/interface.cc

```diff
@@ -255,6 +255,44 @@ namespace aspect
     {}
 
 
+    template <int dim>
+    MaterialModelInputs<dim>::MaterialModelInputs(const FEValuesBase<dim,dim> &fe_values,
+                                                  const typename DoFHandler<dim>::active_cell_iterator *cell_x,
+                                                  const Introspection<dim> &introspection,
+                                                  const LinearAlgebra::BlockVector &solution_vector)
+      :
+      position(fe_values.get_quadrature_points()),
+      temperature(fe_values.n_quadrature_points, numbers::signaling_nan<double>()),
+      pressure(fe_values.n_quadrature_points, numbers::signaling_nan<double>()),
+      pressure_gradient(fe_values.n_quadrature_points, numbers::signaling_nan<Tensor<1,dim> >()),
+      velocity(fe_values.n_quadrature_points, numbers::signaling_nan<Tensor<1,dim> >()),
+      composition(fe_values.n_quadrature_points, std::vector<double>(introspection.n_compositional_fields, numbers::signaling_nan<double>())),
+      strain_rate(fe_values.n_quadrature_points, numbers::signaling_nan<SymmetricTensor<2,dim> >()),
+      cell(cell_x)
+    {
+      // Populate the newly allocated arrays
+      fe_values[introspection.extractors.temperature].get_function_values (solution_vector, this->temperature);
+      fe_values[introspection.extractors.velocities].get_function_values (solution_vector, this->velocity);
+      fe_values[introspection.extractors.pressure].get_function_values (solution_vector, this->pressure);
+      fe_values[introspection.extractors.pressure].get_function_gradients (solution_vector, this->pressure_gradient);
+      fe_values[introspection.extractors.velocities].get_function_symmetric_gradients (solution_vector,this->strain_rate);
```

Comment:
I am thinking of a case where some of the other material model outputs (such as density) may depend on strain rates. In which case those material models would fail if strain_rate is not passed in.